### PR TITLE
Blog: Using MQTT Sparkplug B with Node-RED

### PR DESCRIPTION
**Pull request by sumitshinde-84**
## Checklist

 - [ ] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] I have considered the performance impact of these changes
 - [ ] Suitable unit/system level tests have been added and they pass
 - [ ] Documentation has been updated

### CODE REVIEW

File: src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md
Comment (ZJvandeWeg) [inline src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md:14]:
```suggestion
Connected devices are creating a lot of data but without a standard format, managing this data can be tricky. MQTT helps with messaging, though MQTT is flexible and doesn't standardize message formats. MQTT Sparkplug B solves this problem by providing a clear, standardized format for data. In this guide, we’ll show you how to use MQTT Sparkplug B with Node-RED to make managing your device data easier and more organized.
```

File: src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md
Comment (ZJvandeWeg) [inline src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md:20]: I think a key point is that it's optimized in terms of structure for sensor readings and device data being published. It's not a standard for pushing changes to a device or sensor.

**Comment by ZJvandeWeg** [comment]
@Steve-Mcl I think you have more experience than I have with Sparkplug B, can you review? If not, that's fine, and I'll research a bit

File: src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md
Comment (Steve-Mcl) [inline src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md:82]:
```suggestion
### Prerequisite
```

File: src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md
Comment (Steve-Mcl) [inline src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md:86]:
```suggestion
- [node-red-contrib-mqtt-sparkplug-plus](https://flows.nodered.org/node/node-red-contrib-mqtt-sparkplug-plus): Install this Node-RED package for Sparkplug B support via palette manager.
```

File: src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md
Comment (Steve-Mcl) [inline src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md:128]:
```suggestion
5. Enable the "Send Birth Immediately" option. This ensures that a Birth message (DBIRTH) is sent immediately upon deployment and connection to the MQTT broker. Note that enabling this option will send the `DBIRTH` message when the device node connects, but an `NBIRTH` message will be sent successful connection of mqtt sparkplug out node if you are using.
```

File: src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md
Comment (Steve-Mcl) [inline src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md:137]:
```suggestion
Once you deploy the flow and all devices connect to the MQTT broker, the system automatically send a `DBIRTH` message as soon as each device within the node connects, signalling that the device is ready for data transmission.
```

**Review by Steve-Mcl — CHANGES_REQUESTED** [review]
A few typos and suggestions.

Please do not accept my suggestions if they do not make sense (I am not fully up to speed on Sparkplug.


NOTE: There are about 20+ places there `sparkplug` should be written `Sparkplug` - the spec always refers to `Sparkplug` with a capital "S" - please consider updating the rest of the doc accordingly.

PS, great doc 👍

File: src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md
Comment (Steve-Mcl) [reply src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md:14]: NB: A counter suggestion was offered in my review here: https://github.com/FlowFuse/website/pull/2478#discussion_r1724909709

File: src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md
Comment (Steve-Mcl) [reply src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md:20]: NB: A counter suggestion was offered in my review here: https://github.com/FlowFuse/website/pull/2478#discussion_r1724915459

Comment (Steve-Mcl) [reply src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md:20]: "improves data consistency" didnt "feel right"

In terms of data, consistency is often related to timing and factor not necessarily related to "structure".

Comment (Steve-Mcl) [reply src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md:20]: This re-write removes the "simplifies things" and sticks to the official line from ecplise.  the "simplifies things" is touched on in the next sentence so this was somewhat superfluous IMHO.

**Comment by sumitshinde-84** [comment]
> accordingly



> A few typos and suggestions.
> 
> Please do not accept my suggestions if they do not make sense (I am not fully up to speed on Sparkplug.
> 
> NOTE: There are about 20+ places there `sparkplug` should be written `Sparkplug` - the spec always refers to `Sparkplug` with a capital "S" - please consider updating the rest of the doc accordingly.
> 
> PS, great doc 👍

Thanks a lot, @Steve-Mcl, for the review. You made a huge improvement! I’ll also make sure to update `sparkplug` as you suggested.

File: src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md
Comment (Steve-Mcl) [reply src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md]: outdated

File: src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md
Comment (Steve-Mcl) [reply src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md]: outdated

**Review by Steve-Mcl — APPROVED** [review]
Looks good to me, but please do not merge until @ZJvandeWeg has also reviewed.

**Comment by sumitshinde-84** [comment]
Yes, we also need to update the images and push the tile before merging. I have created an art request for it: https://github.com/FlowFuse/website/issues/2487.

File: src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md
Comment (ZJvandeWeg) [inline src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md:7]: @sumitshinde-84 Please make a tile

Comment (sumitshinde-84) [reply src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md]: Done !

Comment (sumitshinde-84) [reply src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md]: we need to replace the diagrams !

Comment (Steve-Mcl) [reply src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md]: > we need to replace the diagrams !

Are you saying the article wasnt 100% ready @sumitshinde-84?

FWIW, i typically switch a PR to draft if not ready to go.

Comment (sumitshinde-84) [reply src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md]: The article is ready; it just needs the diagram updated with the diagrams Yndira will recreate.

Comment (sumitshinde-84) [reply src/blog/2024/08/using-mqtt-sparkplugb-with-node-red.md]: I created those rough diagrams so that Yndira can understand the type of diagram we need.

**Pull request merged**
